xoa/regrid.py

```diff
@@ -331,6 +331,59 @@ def extrap1d(da, dim, mode, **kwargs):
 extrap1d.__doc__ = extrap1d.__doc__.format(**locals())
 
 
+def isoslice(da, values, isoval, dim=None, **kwargs):
+    """Extract data from var where values==isoval
+
+    Parameters
+    -----------
+    da: xarray.DataArray
+          array from which the data are extracted
+    values: array_like
+          array on which a research of isoval is made
+    isoval: Float
+          value of interest on which we perform research in values array
+    dim   : str
+          dimension shared by da and values on which the slice is made
+    Return
+    ------
+    isovar : array_like
+            Sliced array based on data where values==isoval
+
+    Example
+    -------
+
+    Let's define depth and temperature variables both in 3 dimensions (i,j,k)
+    where i and j are horizontal dimension and k the vertical one.
+
+    dep_at_t20 = isoslice(dep,temp,20)   # depth at temperature=20°C
+    temp_at_z15 = isoslice(temp,dep,-15) # temperature at depth=-15m
+
+
+    """
+
+    if dim is None:
+        dim = xcoords.get_zdim(da,errors="ignore")
+
+    assert dim in da.dims
+    assert dim in values.dims
+
+    da_out = xr.apply_ufunc(
+        interp.isoslice,
```

Comment:
I think a wrapper of `interp.isoslice` is probably better than a direct call to this function to convert dask arrays to numpy arrays and to make sure they are if `float64` type

Reply:
Are you thinking of a new wrapper ?
Why should we convert dask arrays to numpy arrays ? 
There should rather be a check that inputs arrays are Xarray ?

Because with the existing one (`_wrapper1d_`) we "loose" the feature of `xarray.apply_ufunc` which move all core dimensions to the last axes of inputs variables (which is useful because our core function `interp.isoslice` work on the last dimension)

Reply:
The `interp.isoslice` function does not accept dask arrays so they must be converted before being used.
The existing wrapper makes the supposition that the core dimension est the last one and merges other dimensions making sure the size is at least one. It may also work with `regrid.isoslice` without the need to write another wrapper.

Reply:
`interp.isoslice` does not accept dask arrays but `xarray.apply_ufunc` does the conversion for us ? 
I am asking becasue it's working like that for now (without a wrapper) : I just pass the function `interp.isoslice` without worrying about converting my arrays and without move my core dimension to the last position.

Maybe i miss something there ?

If we would like to use `interp.isoslice` in standalone we would need to convert arrays in numpy.

Reply:
I performed some tests with `temp.chunk({"eta_rho": 5})`:

- a guvectorized function accepts dask arrays unlike a numba.njited function, so it works with `interp.isoslice` even with chunks (apply_ufunc does convert dask arrays to numpy arrays), which is good to know
- float32 ar automatically converted,
- it seems you must remove `output_dtypes` otherwize it fails with chunks.

Do you confirm?

Reply:
Thanks for the tests ! So i did the same. 
You mean you must remove `output_sizes` in  `dask_gufunc_kwargs`  ? 
i didn't use `output_dtypes` but maybe we should ?

Reply:
Oups, yes, it's output_sizes.